src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
+        FieldDefinition.RangeValidator soonValidator = new FieldDefinition.RangeValidator("soonValidator", "between now and 2 days from now", "aaaa",
+                FieldDefinition.RangeType.GTE, "2020-11-04",
+                FieldDefinition.RangeType.LTE, "2020-11-06");
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                        new FieldDefinition("textField", FieldDefinition.ColumnType.String)
+                                .setDescription("is texty").setLabel("text field"),
+                        new FieldDefinition("booleanField", FieldDefinition.ColumnType.Boolean)
+                                .setLabel("Boolean Field")
+                                .setRequired(true)
+                                .setURL("fake/list-def.url"),
+                        new FieldDefinition("attachmentField", FieldDefinition.ColumnType.Attachment)
+                                .setHidden(true),
+                        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal).
+                                setScale(1234).setFormat("0.####"),
+                        new FieldDefinition("intField", FieldDefinition.ColumnType.Integer)
+                                .setMvEnabled(true)
+                                .setConditionalFormats(List.of(new ConditionalFormat(List.of(gtFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                                .setValidators(List.of(soonValidator))));
+        dgen.createDomain(createDefaultConnection(), "SampleSet");
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", sampleType);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        String roundTripSampleType = "testSampleTypeWithImportedFields";
+        File exportFile = domainFormPanel.clickExportFields();
+        CreateSampleTypePage sampleTypeCreatePage = CreateSampleTypePage.beginAt(this, getProjectName());
+        sampleTypeCreatePage.setName(roundTripSampleType);
+        sampleTypeCreatePage.getFieldsPanel()
+                .setInferFieldFile(exportFile);
+        sampleTypeCreatePage.clickSave();
+
+        DomainDesignerPage sampleTypeDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", roundTripSampleType);
+        File roundTripFile = sampleTypeDesignerPage.fieldsPanel().clickExportFields();
+        List<PropertyDescriptor> roundTrippedFields = getFieldsFromExportFile(roundTripFile);
+
+        for (PropertyDescriptor intendedField : dgen.getColumns())
+        {
+            String fieldName = intendedField.getName();
+            if (!fieldName.equals("name"))  // don't expect name col to export from sampletype, it is special
```

Comment:
The name field for sample type is special in that we add it automatically when the sample type is first saved. The export feature only exports whatever is "in view" for the domain form panel when that button is clicked. So in the sample type case, if you click export before initially saving the sample type, you probably would not get a "name" field, but if you save and then go back into the sample type and then export, you will.

Reply:
I've updated the test to get the fields from the domain-create response, instead of the ones I sent in.
This means I get back name and key fields in sampleType and list, respectively- and I validate export/import on all of them.

Reply:
That seems dangerous. If our API helper isn't setting some property correctly, it wouldn't be in the create response and we wouldn't end up validating it.

Reply:
OK, I'll also validate here that everything we send in comes back as well

Reply:
Doing so shows that the 'name' field isn't on the sampleType when it is created.
It's also not shown on the UI, not exported, cannot be imported, cannot be added via the UI... but the remoteAPI fails if no 'name' field is supplied when creating the sampleType